ucp/_libs/core.pyx

```diff
@@ -241,28 +242,41 @@ cdef class ApplicationContext:
         # For now, a application context only has one worker
         ucp_worker_h worker
         int epoll_fd
-        object all_epoll_binded_to_event_loop
+        object event_loops_binded_for_progress
+        object progress_tasks
         object config
         bint initiated
+        bint blocking_progress_mode
 
-    def __cinit__(self, config_dict={}):
+    def __cinit__(self, config_dict={}, blocking_progress_mode=None):
         cdef ucp_params_t ucp_params
         cdef ucp_worker_params_t worker_params
         cdef ucs_status_t status
-        self.all_epoll_binded_to_event_loop = set()
+        self.event_loops_binded_for_progress = set()
+        self.progress_tasks = []
         self.config = {}
         self.initiated = False
 
+        if 'UCXPY_NON_BLOCK_MODE' in os.environ:
+            print("UCXPY_NON_BLOCK_MODE")
```

Comment:
Can we try to keep naming more consistent? Everywhere else it's referred to as "blocking", I would expect this to be `UCXPY_NON_BLOCKING_MODE` as well.

Reply:
Good point :)